gym/spaces/box.py

```diff
@@ -78,16 +78,19 @@ def __init__(
 
         # determine shape if it isn't provided directly
         if shape is not None:
-            shape = tuple(shape)
-        elif not np.isscalar(low):
-            shape = low.shape  # type: ignore
-        elif not np.isscalar(high):
-            shape = high.shape  # type: ignore
+            shape = tuple(int(dim) for dim in shape)
+        elif isinstance(low, np.ndarray):
+            shape = low.shape
+        elif isinstance(high, np.ndarray):
+            shape = high.shape
+        elif np.issubdtype(type(low), np.integer) and np.issubdtype(
```

Comment:
Shouldn't we also allow `low` and `high` to be floats/other numeric types? I suspect that in this version, `Box(low=0.5, high=1.2)` would crash, which is not what we want. I think checking `np.isscalar` for this branch was actually appropriate